Sources/LeafKit/LeafSerialize/LeafSerializer.swift

```diff
@@ -70,11 +63,10 @@ internal struct LeafSerializer {
     private mutating func serialize(_ conditional: Syntax.Conditional) throws {
         evaluate:
         for block in conditional.chain {
-            let evaluated = try self.resolveAtomic(parameters: block.condition.expression())
-            if evaluated.bool ?? !evaluated.isNull {
-                try serialize(body: block.body)
-                break evaluate
-            }
+            let evaluated = try resolveAtomic(block.condition.expression())
+            guard (evaluated.bool ?? false) || (!evaluated.isNil && evaluated.celf != .bool) else { continue }
+            try serialize(body: block.body)
+            break evaluate
```

Comment:
Clarified - If the conditional expression is *any non-optional value*, or an explicit true bool